Bug 1210314 - APP Settings implementation

TatianaMigacheva opened the pull request.

harman-ankitgupta committed:
Bug 1210314 - APP Settings implementation

autolander commented:
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f00a0491c7751609191650a5401ec38db874e333

davidflanagan commented on apps/settings/elements/about.html:
```diff
@@ -58,30 +58,12 @@ <h2 data-l10n-id="softwareUpdates"></h2>
 
       <header>
-        <h2 data-l10n-id="aboutBrowserOS"></h2>
+        <h2 data-l10n-id="aboutFirefoxOSTitle"></h2>
```
Don't change this, please. "FirefoxOS" is a trademark and is not open source. Anyone can use this code, but only those with licensing agreements with Mozilal can use "Firefox" and "FirefoxOS". That's why you sometimes see things like "aboutBrowserOS" in our code.

davidflanagan commented on apps/settings/elements/wifi-knowNetworks.html:
```diff
@@ -0,0 +1,23 @@
+<element name="wifi-knowNetworks" extends="section">
```
shouldn't the filename and name attribute on this line be knownNetworks, not knowNetworks?

Ashish-Developer replied:
Will correct this typo.

davidflanagan commented on apps/settings/locales/device_type/phone/settings.device.en-US.properties:
```diff
@@ -2,7 +2,7 @@
 # device, but don't use a simple copy and paste: string ID is the same, but
 # the device type is different in each file.
-change-device-name = Change phone name
+change-device-name = Rename My Device
```
A general rule for localization files like this is that you are not allowed to change existing strings. If the text of the string has to change, you must create a new string id.  So in this case you could use change-device-name-2 for the new string.  Then, of course, any html or JS that uses 'change-device-name' would have to change to 'change-device-name-2'.

If you change the text without changing the id, localizers may not know that a new translation is needed.

davidflanagan commented on apps/settings/locales/device_type/phone/settings.device.en-US.properties:
```diff
@@ -12,9 +12,12 @@ findmydevice-active-tracking = Your phone is being tracked by another device thr
 performanceAndUsageDataInfo = Help us improve {{brandShortName}} by sharing data about your phone and the way you use it.
+# Add new with Help to improve firefox OS
+helpToImproveFirefoxOS =Help to improve Firefox OS
+helpToImproveFirefoxOSMessage=Help us improve Firefox OS by sharing data about your phone and the way you use it.
```
Note the {{brandShortName}} just above these strings. Use that instead of saying "Firefox OS". That's how we deal with releases where the trademark is not licensed.

davidflanagan commented on apps/settings/locales/settings.en-US.properties:
```diff
@@ -63,9 +65,9 @@ shortStatus-connected        = Connected
+wifi-sleep-description=Enable to automatically turn off Wi-Fi during sleep to save battery power.
 manageNetworks=Manage Networks
-wifi-bad-credentials-title=Failed connecting to Wi-Fi
+wifi-bad-credentials-title=Failed Connecting to Wi-Fi
```
Even capitalization changes can't be made without changing the key.  Our UX team has always preferred lower-case for strings like this, however. If the Litehouse specs say otherwise, please double-check with Mozilla's UX team before making string changes like that.

davidflanagan commented on apps/settings/locales/settings.en-US.properties:
```diff
@@ -74,8 +76,9 @@ wifi-bad-credentials-confirm=Unable to connect to {{ssid}}, authentication faile
 manageNetworks-header={{manageNetworks}}
 manageCertificates=Manage certificates
-manageCert=Manage Cert.
+manageCert=Manage Certificates
```
Here's another case were you can't change a string like this.  I'm going to stop commenting on these, but it is a general rule that you can't do this.  

In this particular case, it seems clear to me that our UI needed a very short string for something to fit. So by making the string longer, you probably broke something.

davidflanagan commented on apps/communications/contacts/js/export/bt.js:
```diff
@@ -15,7 +15,7 @@ var ContactsBTExport = function ContactsBTExport() {
 
   var _hasDeterminativeProgress = function btex_hasDeterminativeProgress() {
-    return false;
+    return true;
```
I've got no idea what this function is for, but I'm surprised that you can safely change the return value without changing anything else...

davidflanagan commented on apps/communications/contacts/js/views/list.js:
```diff
@@ -1675,11 +1672,50 @@ contacts.List = (function() {
         callback(uuid);
       });
     }
+    }
```
something is wrong with the indentation here.

davidflanagan commented on apps/communications/contacts/locales/contacts.en-US.properties:
```diff
@@ -110,9 +112,9 @@ exportContactsButton    = Export Contacts
-btExport-title         = Export to Bluetooth
+simExport-title        = Exporting
+memoryCardExport-title = Exporting
+btExport-title         = Exporting
```
You can't changes strings without also changing the l10nId key. It messes up our localizers if you do that.

Also, it seems premature to me to be making this kind of string changes.  The UX we have from Litehouse so far seems like wireframes. I would expect final text to come along with final visual specs...  Let's not change too many strings now unless you've double-checked that these are what is really wanted.  I assume that some of the stuff in the wireframes is just placeholder text.

davidflanagan commented on apps/communications/contacts/style/app.css:
```diff
@@ -513,6 +514,7 @@ li button.icon-find-and-merge:before {
 
 nav[data-type="scrollbar"] {
   z-index: 6;
+  display: none;
```
If this is being hidden for non-touch, consider just removing it completely instead.

davidflanagan commented on apps/communications/contacts/style/contacts.css:
```diff
@@ -426,7 +425,7 @@ gaia-header [data-icon] {
 #view-contact-form .view-body-inner,
 #ice-settings .view-body-inner {
-  background: #f4f4f4;
+  background: #ffffff;
```
Why are you making minor color adjustments?  I thought we only had wireframes. Are there already visual design specs that you're following here?

Ashish-Developer replied:
For some cases we received visual as well, thats why we did this changed.
Complete Visual design specs implementation going on right now.

davidflanagan commented on apps/communications/contacts/style/contacts.css:
```diff
@@ -1118,3 +1156,260 @@ html[dir="rtl"] #custom-tag-reset {
+}
+
+#search-image {
+  background-image: url("images/search.svg");
```
We don't have any existing .svg images in the app. I doubt we want to start with that now.  There are a lot of svg icons embedded in our icon font, however, and maybe there is already something there that you can use.  I don't have the details myself, unfortunately.

davidflanagan commented on apps/communications/contacts/style/contacts.css:
```diff
@@ -1118,3 +1156,260 @@ html[dir="rtl"] #custom-tag-reset {
+  border-radius: 2rem;
+  height: 4rem;
+}
+
```
Lots of styles for role="dialog". Did you just copy these from one of the shared files? That doesn't seem right. If style changes are needed for dialogs, other apps will probably need them too.  So we should change them in the shared location.

davidflanagan commented on apps/communications/contacts/style/images/search.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 30 30"><path fill="#aaa" d="M23.385 21.494c.266.266.057.903-.465 1.426s-1.16.729-1.426.465l-4.041-4.039c-1.01.656-2.21 1.047-3.505 1.047-3.561 0-6.448-2.885-6.448-6.446 0-3.56 2.887-6.446 6.448-6.446 3.562 0 6.449 2.887 6.449 6.446 0 1.297-.393 2.499-1.051 3.509l4.041 4.039h-.002v-.001zm-9.437-12.459c-2.712 0-4.914 2.198- ...
```
See if there is something in the icon font that you can use instead of this...

davidflanagan commented on apps/settings/js/findmydevice.js:
```diff
@@ -153,13 +153,23 @@ define('findmydevice', ['modules/settings_utils', 'shared/settings_listener'
 
       if (checkbox.checked === false) {
+        finddevice.textContent = "Disabled";
+        findmydevice.textContent= "Disabled";
```
These are unlocalized strings. You can't do this.

davidflanagan commented on apps/settings/js/findmydevice.js:
```diff
@@ -153,13 +153,23 @@ define('findmydevice', ['modules/settings_utils', 'shared/settings_listener'
         SettingsHelper('findmydevice.enabled').set(true, function() {
-          checkbox.disabled = false;
+            finddevice.textContent = "Enabled";
+            findmydevice.textContent = "Enabled";
```
same problem here.

Vinay18Mishra committed:
Bug 1210314 - Settings: Fix on visual design's changes

Changes as per Visual Design, review comments incorporated as @egor, @sergey,@david and @ashish

Commit:
Merge pull request #20 from Vinay18Mishra/bug-1210314-settings

Bug 1210314 - Settings: Initial Change on Visual Design

autolander commented:
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fd6102b6d57e65dcbf61a3afa3aec604a8dcf68e

Commit:
Bug 1210314 - Settings implementation

harman-ankitgupta committed:
Bug 1210314 - Settings

Commit:
Merge pull request #37 from ankitgupta020/bug-1210314-settings

Bug 1210314 Settings

autolander commented:
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5085f9ff5322ec1fb6cbecb55d382180c0daa213

Commit:
Merge pull request #38 from inrkapoor020/Bug-1210314-settings

Bug 1210314 - Settings implementation : Fix JS Lint errors

autolander commented:
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b8a442e3fe483dd935e3baec94bc4a034f112414

TatianaMigacheva commented:
This PR has been closed because RedSquare project has been canceled. All changes which were done by Bug#1210314 will be attached as Bugzilla patch.
